skbio/stats/distance/_base.py

```diff
@@ -715,28 +767,31 @@ def from_iterable(cls, iterable, metric, key=None, keys=None):
 
         Notes
         -----
-        Symmetry and hollowness are assumed when calculating the distances via
-        `metric`. Therefore, distances are only computed for the strictly
+        When `validate=True`, distances are only computed for the strictly
         upper/lower triangle.
 
         """
-        iterable = list(iterable)
-        if key is not None and keys is not None:
-            raise ValueError("Cannot use both `key` and `keys` at the same"
-                             " time.")
+        if validate:
```

Comment:
This `if` statement is inverted from what it should be, we want to call `super` if `validate=True`, else only compute lower triangle.